themes/lucky/lucky.theme.sh

```diff
@@ -0,0 +1,59 @@
+#! bash oh-my-bash.module
+
+_omb_deprecate_const 20000 red "$_omb_prompt_brown" "${_omb_deprecate_msg_please_use/'%s'/_omb_prompt_brown}"
+_omb_deprecate_const 20000 blue "$_omb_prompt_navy" "${_omb_deprecate_msg_please_use/'%s'/_omb_prompt_navy}"
+_omb_deprecate_const 20000 green "$_omb_prompt_green" "${_omb_deprecate_msg_please_use/'%s'/_omb_prompt_green}"
+_omb_deprecate_const 20000 purple "$_omb_prompt_purple" "${_omb_deprecate_msg_please_use/'%s'/_omb_prompt_purple}"
+
+SCM_GIT_SHOW_MINIMAL_INFO=true
+SCM_THEME_PROMPT_PREFIX=""
+SCM_THEME_PROMPT_SUFFIX=""
+
+L_BLUE=$_omb_prompt_navy
+L_CYAN=$_omb_prompt_teal
+L_GREEN=$_omb_prompt_green
+L_NORMAL=$_omb_prompt_normal
+L_PURPLE=$_omb_prompt_purple
+L_RED=$_omb_prompt_brown
+L_WHITE=$_omb_prompt_white
+
+L_BOLD_RAD=$_omb_prompt_bold_brown
+L_BOLD_YELLOW=$_omb_prompt_bold_olive
+
+L_PREFIX=$SCM_THEME_PROMPT_PREFIX
+L_SUFFIX=$SCM_THEME_PROMPT_SUFFIX
+
+# reset git_prompt_minimal_info
+function git_prompt_minimal_info {
+  local ref
+  local status
+  local git_status_flags=('--porcelain')
+  SCM_STATE=${L_GREEN}${SCM_THEME_PROMPT_CLEAN}
+
+  if _omb_prompt_git_status_enabled; then
+    # Get the branch reference
+    ref=$(git_clean_branch) ||
+      ref=$(_omb_prompt_git rev-parse --short HEAD 2>/dev/null) || return 0
+    SCM_BRANCH=${SCM_THEME_BRANCH_PREFIX}${ref}
+
+    # Get the status
+    [[ "${SCM_GIT_IGNORE_UNTRACKED}" = "true" ]] && git_status_flags+='-untracked-files=no'
+    status=$(_omb_prompt_git status ${git_status_flags} 2>/dev/null | tail -n1)
+
+    if [[ -n ${status} ]]; then
+      SCM_DIRTY=1
+      SCM_STATE=${L_RED}${SCM_THEME_PROMPT_DIRTY}
+    fi
+
+    # Output the git prompt
+    echo -e " ${L_WHITE}on ${L_BLUE}git:${L_PURPLE}${SCM_BRANCH}${SCM_STATE}"
+  fi
```

Comment:
So as far as I understand it correctly, the difference from the original `git_prompt_minimal_info` in `lib/omb-prompt-base.sh` is these color codes and the prefix "` on `"?

I have a naive question. Couldn't that be achieved just by modifying the global variables `SCM_THEME_PROMPT_DIRTY`, `SCM_THEME_PROMPT_PREFIX`, and `SCM_THEME_PROMPT_SUFFIX`? If so, it should be implemented by the global variables instead of defining a new function.

Reply:
I checked the `lucky` theme on my local host. We can just call `scm_prompt_info` with custom `SCM_THEME_*` variables to achieve the same behavior (unless you want to allow users to change `SCM_THEME_PROMPT_{CLEAN,DIRTY,BRANCH_PREFIX}` after loading the theme). I pushed the changes to the PR branch  (i.e. `themes-lucky`) in commit 2c68bf61bba61bbbe623a44a86a389684fc9e09b. @meizikeai It's supposed to keep the result, but could you pull the branch and check if the result is what you expect?

I see some of your changes in `_omb_theme_lucky_git_prompt_minimal_info` are valuable. I applied them to `git_prompt_minimal_info` in `lib/omb-prompt-base.sh` in commit fa7e58ceab55196f6adb7826ac6b41896392df33.

Reply:
Thanks!